artifacts.go

```diff
@@ -58,7 +58,13 @@ func artifactsHandler(indexer Indexer, cacheTime time.Duration) func(w http.Resp
 			return
 		}
 
+		aPackage := packageList[0]
+		if aPackage.RemoteResolver() != nil {
+			aPackage.RemoteResolver().RedirectArtifactsHandler(w, r, aPackage)
+			return
+		}
+
 		cacheHeaders(w, cacheTime)
-		packages.ServePackage(w, r, packageList[0])
```

Comment:
Nit. `ServePackage` could include the logic to serve local or remote packages, same thing with `ServeFile` and `ServeSignature`.
This way it is easier to remember that we need to take this into account if we add a new handler that uses these helpers, or a new `ServeSomething` helper.

Reply:
I tried that approach at first but decided to hide the logic of local/remote behind filesystem interfaces. I decided to revert when I had to implement the `ReadSeekCloser` interface.

I understand that you propose...

```go
		aPackage := packageList[0]
		if aPackage.RemoteResolver() != nil {
			aPackage.RemoteResolver().RedirectArtifactsHandler(w, r, aPackage)
			return
		}

		cacheHeaders(w, cacheTime)
		packages.ServeLocalPackage(w, r, packageList[0], packageList[0].BasePath)
```

... to move `if aPackage.RemoteResolver() != nil {` logic into `packages.ServeLocalPackage`, right? What do you recommend to do with `cacheHeaders`? Should this function call be also moved to `packages.ServeLocalPackage`?

Reply:
The code here could stay the same.
```
cacheHeaders(w, cacheTime)
packages.ServePackage(w, r, packageList[0])
```
`ServePackage` would include the new logic to do the redirect if the package passed as argument includes a remote resolver.

Reply:
Well, unless we need different cache times for redirects.

Reply:
Fixed.